e2e/scripts/aws-s3-client.js

```diff
@@ -0,0 +1,180 @@
+const {
+    S3Client,
+    PutObjectCommand,
+    GetObjectCommand,
+    ListObjectsV2Command,
+    DeleteObjectCommand,
+    HeadObjectCommand
+} = require('@aws-sdk/client-s3')
+const {STSClient, AssumeRoleCommand} = require('@aws-sdk/client-sts')
+
+class SecureS3Client {
+    constructor(options = {}) {
+        this.roleArn = options.roleArn
+        this.roleSessionName = options.roleSessionName || 'LocalDev'
+        this.region = options.region || 'us-east-1'
+        this.readOnly = options.readOnly
+        this.credentials = null
+    }
+
+    async initialize() {
+        if (this.roleArn) {
+            await this._assumeRole()
+        }
+
+        // Create S3 client after role assumption
+        this.s3 = new S3Client({
+            region: this.region,
+            credentials: this.credentials
+        })
+
+        console.log(`🔐 Using ${this.readOnly ? 'READ-ONLY' : 'READ-WRITE'} access`)
+    }
+
+    async _assumeRole() {
+        try {
+            const sts = new STSClient({region: this.region})
+            const command = new AssumeRoleCommand({
+                RoleArn: this.roleArn,
+                RoleSessionName: this.roleSessionName,
+                DurationSeconds: 3600,
+                ExternalId: 'developer-access'
```

Comment:
is this id fixed? Should we create constant at top?

Reply:
Moved it to env vars